src/server/hotkey_collector.h

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2017, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include <functional>
+#include <dsn/utility/error_code.h>
+#include <rrdb/rrdb_types.h>
+#include <gtest/gtest_prod.h>
+#include <readerwriterqueue/readerwriterqueue.h>
+#include <dsn/dist/fmt_logging.h>
+#include <dsn/utility/flags.h>
+#include <dsn/dist/replication/replica_base.h>
+#include "base/pegasus_utils.h"
+
+namespace pegasus {
+namespace server {
+
+DSN_DECLARE_int32(data_capture_hash_bucket_num);
+
+typedef std::shared_ptr<std::unordered_map<int, int>> thread_queue_map;
+
+class hotkey_coarse_data_collector;
+class hotkey_fine_data_collector;
+
+enum class collector_state
+{
+    STOP,   // data has been cleared, ready to start
+    COARSE, // is running corase capture and analyse
+    FINE,   // is running fine capture and analyse
+    FINISH  // capture and analyse is done, ready to get result
+};
+
+// `hotkey_collector` is responsible to find the hot keys after the partition was detected to be
+// hot. In general, there are 2 collectors once hotkey detection is enabled: one for read, the other
+// for write.
+class hotkey_collector : public dsn::replication::replica_base
+{
+public:
+    hotkey_collector(dsn::apps::hotkey_type::type hotkey_type,
+                     dsn::replication::replica_base *r_base);
+    // after receiving START RPC, start to capture and analyse
+    bool init();
```

Comment:
Call it `start` and rename `clear` to `stop`. Keep the names aligned with the operation types.